delphi/app/plugins/graph_plugin.py

```diff
@@ -0,0 +1,152 @@
+from delphi.app import data_analyzer
+from collections import namedtuple
+
+from delphi.app.graph import FlowNetwork
+from delphi.app.pki_model import PKI
+
+
+def find_path(start_node, flow, people):
+    """
+    find_path is a utility function that returns all that nodes that are able to carry flow from a specific source
+    node
+    :param start_node: The source node to calculate paths for
+    :param flow: List of all nodes in the network with positive flow
+    :param people: The amount of people that must be carried from the source node
+    :return: A list of all possible nodes that contribute to viable paths for start_node
+    """
+    path_list = [start_node]
+    for node in path_list:
+        current_node = node
+        for edge_check in flow:
+            if edge_check.start == current_node and edge_check.end != 'ta':
+                if edge_check.flow >= people / 2:
+                    if edge_check.end not in path_list:
+                        path_list.append(edge_check.end)
+    return path_list
+
+
+class GraphPlugin(data_analyzer.DataAnalyzerPlugin):
+    RoomData = namedtuple('RoomData', ('Room', 'Count'))
+    NodeConstraint = namedtuple('NodeConstraint', ('node',))
```

Comment:
I think you may need more than just the node. You might need to take in the node id and status that is requested.

Reply:
I'm gonna go back and change that parameter to node_id  to be more clear. Also, with regards to the status, a node can only be in two states (enabled and disabled) . So if the model is receiving a request to update status, its safe to assume that the status being requested is the opposite of the state it's in, correct? For example, if a node is disabled and a request is made to update the status of that node, it is same to assume it needs to be enabled. I can see how adding a status parameter might be a good check though. Let me know what you're thinking...

Reply:
I don't see a way that this wouldn't work, so I think it'll be ok